tests/Persisters/Serializers/NativeTest.php

```diff
@@ -42,19 +45,41 @@ public function build() : void
         $this->assertInstanceOf(Native::class, $this->serializer);
         $this->assertInstanceOf(Serializer::class, $this->serializer);
     }
-    
+
     /**
      * @test
      */
     public function serializeUnserialize() : void
     {
         $data = $this->serializer->serialize($this->persistable);
-        
+
         $this->assertIsString($data);
 
         $persistable = $this->serializer->unserialize($data);
 
         $this->assertInstanceOf(DummyClassifier::class, $persistable);
         $this->assertInstanceOf(Persistable::class, $persistable);
     }
+
+    /**
+     * @test
+     */
+    public function deserializeNonObject() : void
+    {
+        $data = serialize(new stdClass());
+        $this->expectException(RuntimeException::class);
+
+        $this->serializer->unserialize($data);
+    }
+
+    /**
+     * @test
+     */
+    public function deserializeNonPersistable() : void
+    {
```

Comment:
Is this 'essentially' a duplicate test as the one above? Could it be factored into one test or perhaps a data provider would be helpful (in which case the method name could be something like deserializeBadData?

Reply:
I copy-pasted and forgot to change `stdClass` to something that is not an object. In any case, combining the two similar methods makes sense.